1-Authentication/7-sign-in-express-mfa/Api/index.js

```diff
@@ -0,0 +1,65 @@
+// Import necessary modules
+const express = require("express");
+const msal = require("@azure/msal-node");
+const app = express();
+const port = 3001;
+const { v4: uuidv4 } = require("uuid");
+const {
+  msalConfig,
+  GRAPH_ME_ENDPOINT,
+  TENANT_SUBDOMAIN,
+  TENANT_ID,
+} = require("./authConfig");
+
+var { fetch } = require("./fetch");
+
+// Middleware to parse incoming JSON requests
+app.use(express.json());
+
+const cca = new msal.ConfidentialClientApplication(msalConfig);
+
+// Function to acquire token using client credentials
+async function getAccessToken(tokenRequest) {
+  try {
+    const response = await cca.acquireTokenOnBehalfOf(tokenRequest);
+    return response.accessToken;
+  } catch (error) {
+    console.error("Error acquiring token:", error);
+    throw error;
+  }
+}
+
+// Define the updateUserInfo endpoint
+app.post("/updateUserInfo", async (req, res) => {
+  try {
+    const tokenRequest = {
+      oboAssertion: req.headers.authorization.replace("Bearer ", ""),
+      authority: `https://${TENANT_SUBDOMAIN}.ciamlogin.com/${TENANT_ID}`,
```

Comment:
```suggestion
      authority: msalConfig.authority,
```

This is brittle, what if my authority is a vanity domain? We should use the `authority` property of the `msalConfig`